tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -1160,28 +1162,67 @@ export class WebGPUBackend extends KernelBackend {
 
   fromPixels(
       pixels: backend_util.PixelData|ImageData|HTMLImageElement|
-      HTMLCanvasElement|HTMLVideoElement,
+      HTMLCanvasElement|HTMLVideoElement|ImageBitmap,
       numChannels: number): Tensor3D {
     if (pixels == null) {
       throw new Error(
           'pixels passed to tf.browser.fromPixels() can not be null');
     }
 
+    // TODO(tfjs-optimization): These has chance to be cached and do not update
+    // buffer always.
     const outShape = [pixels.height, pixels.width, numChannels];
+    const size = util.sizeFromShape(outShape);
+    const uniformData = [size, pixels.width, numChannels];
     let imageData = (pixels as ImageData | backend_util.PixelData).data;
 
     if (env().getBool('IS_BROWSER')) {
       if (!(pixels instanceof HTMLVideoElement) &&
           !(pixels instanceof HTMLImageElement) &&
           !(pixels instanceof HTMLCanvasElement) &&
           !(pixels instanceof ImageData) &&
-          !(pixels.data instanceof Uint8Array)) {
+          !(!(pixels instanceof ImageBitmap) &&
+            (pixels.data instanceof Uint8Array)) &&
+          !(pixels instanceof ImageBitmap)) {
         throw new Error(
             'pixels passed to tf.browser.fromPixels() must be either an ' +
             `HTMLVideoElement, HTMLImageElement, HTMLCanvasElement, ImageData` +
             ` or {data: Uint32Array, width: number, height: number}, ` +
             `but was ${(pixels as {}).constructor.name}`);
       }
+
+      if (pixels instanceof ImageBitmap) {
+        const output = this.makeOutputArray(outShape, 'int32');
+        if (!this.fromPixelProgram) {
+          this.fromPixelProgram = new FromPixelsProgram(outShape);
+
+          const {bindGroupLayout, pipeline} = webgpu_program.compileProgram(
+              this.glslang, this.device, this.fromPixelProgram, [], output);
+
+          // TODO(tfjs-optimization): May use direct number instead of uniform
+          // buffer
```

Comment:
what do you mean by use direct number?

Reply:
This means the same way as current workGroupSize did (and generate multiple shaders)

Reply:
Modify the comments.